src/providers/test-result/test-result.ts

```diff
@@ -38,6 +42,14 @@ export class TestResultProvider {
       case TestCategory.EUA2M1:
       case TestCategory.EUAMM1:
         return this.calculateCatAAndSubCategoryTestResult(TestCategory.EUAM1, testData as TestData);
+      case TestCategory.D:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D, testData  as CatDUniqueTypes.TestData);
+      case TestCategory.D1:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D1, testData  as CatD1UniqueTypes.TestData);
+      case TestCategory.DE:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.DE, testData  as CatDEUniqueTypes.TestData);
+      case TestCategory.D1E:
+        return this.calculateCatCDndSubCategoryTestResult(TestCategory.D1E, testData as CatD1EUniqueTypes.TestData);
```

Comment:
These can be handled together; would make the switch a bit cleaner

Reply:
fixed  - nice spot